lib/src/sql/value/value.rs

```diff
@@ -1221,9 +1221,28 @@ impl ops::Div for Value {
 	type Output = Self;
 	fn div(self, other: Self) -> Self {
 		match (self, other) {
-			(Value::Number(v), Value::Number(w)) => Value::Number(v / w),
-			(Value::Datetime(v), Value::Duration(w)) => Value::Datetime(w / v),
-			(v, w) => Value::from(v.as_number() / w.as_number()),
+			(Value::Number(v), Value::Number(w)) => {
+				if w == Number::Int(0) {
+					Value::None
+				} else {
+					Value::Number(v / w)
+				}
+			}
+			(Value::Datetime(v), Value::Duration(w)) => {
+				if w.is_zero() {
+					Value::None
+				} else {
+					Value::Datetime(w / v)
```

Comment:
Btw: I opened a discussion about this "feature" because it seems questionable to implement division by truncating.

https://github.com/surrealdb/surrealdb/discussions/1207